haproxy-healthcheck/check.py

```diff
@@ -0,0 +1,77 @@
+import os
+import socket, requests
+
+from _thread import *
+import threading
+
+print_lock = threading.Lock()
+
+couchdb_servers = os.environ['COUCHDB_SERVERS'].split(',')
+# Example: COUCHDB_SERVERS="couchdb.1,couchdb.2,couchdb.3"
+username = os.environ['COUCHDB_USER']
+password = os.environ['COUCHDB_PASSWORD']
+
+def get_membership_endpoint(couchdb_url):
+    return f'http://{username}:{password}@{couchdb_url}:5984/_membership'
+
+def close_connection(connection):
+    print('Closing connection ...')
+    connection.close()
+
+def send_down_response(conn):
+    print('Sending response: Down (CouchDB cluster not okay) ...')
+    conn.send(b'down\n')
+    close_connection(conn)
+
+def send_up_response(conn):
+    print('Sending response: Up (Couchdb cluster okay) ...')
+    conn.send(b'up\n')
+    close_connection(conn)
+
+def threaded(conn):
+    try:
+        result = []
+        couchdb_server = couchdb_servers[0]
+        try:
+            r = requests.get(get_membership_endpoint(couchdb_server), timeout=1)
+        except Exception as e:
+            print("Exception: ", e)
+        data = r.json()
+
+        all_nodes = data['all_nodes']
+        cluster_nodes = data['cluster_nodes']
+
+        if all_nodes == cluster_nodes and all_nodes is not None:
```

Comment:
Can we also check that the nodes match `couchdb_servers`, so the healthcheck fails before the cluster is set up?

Reply:
Done.